### ISSUE TITLE
Definition of dataset: harmonisation needed

### ISSUE THREAD

**Issue by heidivanparys** [opengeospatial/ogcapi-features#312]
The definition of "dataset" is the one from DCAT. However, ISO/TC 211 uses [another definition](https://www.iso.org/obp/ui/#iso:std:iso:19115:-2:ed-2:v1:en:term:3.6), and ISO/TC 211 maintains a [glossary of terms](https://committee.iso.org/sites/tc211/home/re.html), with the purpose to encourage consistency in the use and interpretation of geospatial terms.

Given the fact that OGC API - Features - Part 1: Core is supposed to become an ISO/TC 211 standard, this is an issue.

Therefore, the following proposal, also submitted via W3C (see https://github.com/w3c/dxwg/issues/1195) and ISO/TC 211.

---

# Background and problem statement

This is a joint proposal for an updated definition for the concept “dataset”, made by the Danish Agency for Digitisation [1] (member of W3C) and the Danish Agency for Data Supply and Efficiency [2] (member of OGC and member of ISO/TC 211).

The Danish government wants to describe data consistently across authorities, processes and IT systems [3]. In order to achieve this, definitions of the same concepts must be aligned where possible, not only within domains but also across domains.

“Dataset” is used in many different domains and is a highly relevant concept these days, e.g. in the context of the European INSPIRE Directive [4] and the PSI Directive [5]. The Agency for Digitisation and the Agency for Data Supply and Efficiency therefore request W3C, OGC and ISO/TC 211 to come to an agreement regarding using the same definition and notes for “dataset” and submit a first draft for discussion.

# Proposal for updated definition and related notes

**dataset**
collection of data that is regarded as a unit

Note 1 to entry: Typically, a dataset is collected for a certain purpose.
Note 2 to entry: Typically, a dataset is described using metadata elements including an identifier and a title.
Note 3 to entry: Typically, a dataset is available for use in one or more representations.
Note 4 to entry: Typically, a dataset is published or curated by a single agent.
Note 5 to entry: Typically, the data in a dataset are related through a common topic.
Note 6 to entry: Typically, the data in a dataset have the same syntactic structure.
Note 7 to entry: Typically, the data in a dataset are managed using the same governance processes.
Note 8 to entry: Typically, the data in a dataset have a shared data provenance.
Note 9 to entry: The arrangement of data in one or more datasets is a decision, based on formal requirements or informal considerations.

| Data catalog specific notes | Geographic information specific notes |
| ---- | ---- |
| Note 10 to entry: In the context of DCAT, a dataset is published or curated by a single agent.                          | Note 10 to entry: In the context of geographic information, a dataset can be a smaller grouping of data which, though limited by some constraint such as spatial extent or feature type, is located physically within a larger dataset. Theoretically, a dataset can be as small as a single feature or feature attribute contained within a larger dataset. |
| Note 11 to entry: In the context of DCAT, a dataset is available for access or download in one or more representations. | Note 11 to entry: In the context of geographic information, a hardcopy map or chart can be considered a dataset.|

In order to highlight the need for all the notes to the definition of “dataset”, hereby a list of examples that are **not** “typical” according to those notes. The examples are not intended to be included in any standard, but are meant as a basis for the discussion. The examples follow the same numbering as the notes above.

Example 1: Dataset that is too large or complex to analyze with the current technologies, but that might be useful when technology evolves.
Example 2: A temporary dataset, such as the result of an SQL query.
Example 3: A planned dataset, that is not yet available or collected.
Example 4: OpenStreetMap data; in general: data collected via crowdsourcing.
Example 5: A use case where data from many different domains (different subjects) is combined to solve a particular problem or need (one purpose).
Example 6: GeoPackage containing vector, raster and styling; EU-dataset where some of the data are INSPIRE-harmonized and some are not.
Example 7: Data collected at the local level in a country, and then aggregated to one dataset containing data for the whole country; data sent in to EEA by the EU member states, and which is then aggregated to one dataset.
Example 8: Dataset containing data having different licences; dataset created by taking subsets from different other datasets.

# References

[1] https://en.digst.dk/
[2] https://eng.sdfe.dk/
[3] THE GOVERNMENT / LOCAL GOVERNMENT DENMARK / DANISH REGIONS. _The digitally coherent public sector_ [online]. Version 1.0. Agency for Digitisation, June 2017. Available from: https://arkitektur.digst.dk/sites/default/files/white_paper_on_a_common_public-sector_digital_architecture_pdfa.pdf
[4] _Directive 2007/2/EC of the European Parliament and of the Council of 14 March 2007 establishing an Infrastructure for Spatial Information in the European Community (INSPIRE)_ [online]. 25 April 2007. 32007L0002. Available from: http://data.europa.eu/eli/dir/2007/2/oj/eng
[5] _Directive (EU) 2019/1024 of the European Parliament and of the Council of 20 June 2019 on open data and the re-use of public sector information_ [online]. 26 June 2019. 32019L1024. Available from: http://data.europa.eu/eli/dir/2019/1024/oj/eng

**Comment by cportele** [comment]
+1 - it would be a good step, if there is a broad consensus of the definition of the term "dataset" and I would be happy to see the definition updated in a future revision of the document to align with an updated definition in W3C / schema.org. 

Personally, I do not really care about the exact definition as long as it is general enough and not incorrect. The main reason for using the DCAT definitions for dataset and distribution were not the definitions themselves, but to express that we want to use the concepts from the web architecture, not a specific view of the geo-community. "dataset" and "distribution" are general terms not specific to geo (unlike "feature"), so it was intentional to not use the definitions from the TC211 glossary.

I am a little worried about the number of notes in the proposal. If this pattern is followed in general, a lot of content could be moved from the main clauses of a standard to the terms and definitions clause (or it could lead to significant repetition / redundancies). In my understanding, the terms and definitions clause is not about the details of a concept, it is about brief definitions that can replace a term in written text, but I will leave that discussion to the terminology experts.

**Comment by heidivanparys** [comment]
> [...] about the exact definition as long as it is general enough and not incorrect

We argue that the current DCAT definition actually is too narrow and excludes entities that should be covered by the concept "dataset". See the comment made by @aidig in https://github.com/w3c/dxwg/issues/1195#issuecomment-566719086 and see also the counterexamples from the proposal.

Regarding the ISO/TC 211 definition, our experience is that most people don't understand the "identifiable" part of the definition.

> I am a little worried about the number of notes in the proposal. If this pattern is followed in general, a lot of content could be moved from the main clauses of a standard to the terms and definitions clause (or it could lead to significant repetition / redundancies). In my understanding, the terms and definitions clause is not about the details of a concept, it is about brief definitions that can replace a term in written text, but I will leave that discussion to the terminology experts.

The reason for adding the notes was to make to it clear that many existing definitions of "dataset" include one or more characteristics in the definition, that - in our view - are _typical_, but not _defining_, for what a dataset is.

However, the same concern has been expressed to us via other channels and we'll have to figure out how to address that concern, while still following good terminology practice.

In short:

1. We need consensus on a definition that is concise and precise (and that can pass the substitution test indeed).
2. We may need another solution for the notes.

**Comment by heidivanparys** [comment]
From https://github.com/w3c/dxwg/issues/1195#issuecomment-566766634 by @rob-metalinkage 

> The second part of the puzzle is how OGC specifications refer to the concept of Dataset

I haven't been involved in OGC for that long, but I was under the impression that OGC often reuses the terminology from ISO/TC 211, and that OGC and ISO/TC 211 usually try to harmonize? From https://www.opengeospatial.org/ogc/glossary/d : "dataset" is not present, but "dataset series" is, and it refers to ISO.

**Comment by ogcscotts** [comment]
@heidivanparys OGC generally tries to follow ISO / TC 211 definitions and is working to replace some of our definitions with those from ISO. See Policy Directive 45: https://portal.opengeospatial.org/public_ogc/directives/directives.php. All that being said, the policy is to use ISO / TC 211 definition when they exist and are appropriate. We can use other sources if those sources are more suitable.

**Comment by cportele** [comment]
@heidivanparys - what @ogcscotts said. In the [first comment above](https://github.com/opengeospatial/ogcapi-features/issues/312#issuecomment-567056866) I explained why we didn't use the ISO definition in this case, but one from W3C.

**Comment by heidivanparys** [comment]
@ogcscotts  @cportele  Thanks for the clarification. So the answer to the question of @rob-metalinkage in https://github.com/w3c/dxwg/issues/1195#issuecomment-566766634 would be that we don't need to dive into all OGC specifications but can safely assume that in (almost) all specifications, the ISO/TC 211 definition will have been used.

**Comment by rob-metalinkage** [comment]
@heidivanparys  - if only it were that easy :-)  The process of harmonisation is an ongoing one, and there is a long history of specs that do not get automatically updated... so the assumption should be that the OGC Architecture Board (OAB) successfully catches and enforces harmonisation in all new specifications proposed.

The second challenge, which is highlighted by this state of things but exists anyway is declaring what terms mean where they are used - again there is a trajectory from ad-hoc term definition, to gathering a glossary of terms - to actually making terms explicit. Terms can only be made explicit if they have unambiguous unique identifiers and a way of resolving those to find the definition.  A human readable approach exists by having definitions in text in documents and scoping use of words to those inline definitions - but this doesnt make it possible, for example, to find which specifications use which definition or make any machine readable statements about the specification using those terms.  This is where the OGC Definitions Server provides a mechanism to have unambiguous URIs and various human- and  machine-readable options - but at this stage we dont have mechanisms in general to map term semantics in documents to these URIs, however we are exploring what we can do to gradually build out a set of relationships. We can then look at supporting improved knowledge access using that data - for example which specs use an ISO compatible definition and which do not.

It would be good, for example, to be able to quickly identify which requirements (and the conformance classes they belong to) rely on a any given definition.   Getting the body of conformance classes and requirement definitions into the OGC Definitions Server is on our backlog of tasks - each of these tasks means establishing a rationale, doing some testing, signing off the characterisation, then automating the connection between the point of truth systems in OGC and the DefServer - and then looking at usefully populating all the cross-references that make sense now the identifiers are available.  NB Guidance from the community about the most useful content is always welcome. 

Secondly we really need to automate linking terminology in documents to the globally scoped canonical URIs for definitions - and this needs to be in the document template and processing chain for new specification and also a retrofitting exercise (possibly viewing the authoritative documents via some annotating service if we cannot change the underlying artifacts).  None of this effort is currently explicitly resourced so progress is a by-product of resourced projects and therefore too slow.

**Comment by cportele** [comment]
Meeting 2019-12-23: This is a bigger topic than just Features and it also is related to liaisons (W3C, TC 211). This should be discussed in the OAB (both the specific question about the dataset definition and the broader questions about the OGC processes for terms/definitions). @cportele to create two issues in the OAB section of the portal. Part 2 is already on the agenda of the next OAB call and we could discuss this topic then, too.

**Comment by cportele** [comment]
Update: There was no time in the OAB meeting, but issue triage is on the agenda of the next OAB meeting.

I move this issue to "waiting for feedback" for now.

**Comment by ReesePlews** [comment]
i replied to DK (@heidivanparys and her colleagues) and 19168-1 PT and i will participate in any discussions about this. 

in my reply i gave some background from the TC 211 TMG standpoint. how long as this (19168-1 dataset def) been used by OGC and in what standards?  i would guess it is as extensive as TC211's use but i suspect TC211s definition may be a bit older. please keep me informed of any pending discussions on this.

**Comment by cportele** [comment]
Thanks, @ReesePlews. We will keep this open for feedback for now and wait for feedback from the OAB before trying to agree on a resolution.

**Comment by cportele** [comment]
The OAB discussed the issue today and here is their guidance for the future definition of "dataset" in OGC:

----
**dataset**
collection of data 

NOTE: published or curated by a single agent, and available for access or download in one or more serializations or formats.

----

(There is a bit more discussion in the OAB meeting notes on the OGC portal.)

My proposal is to adopt this in the 1.0.1 version of the OGC standard marking the definition as adapted from DCAT. We should discuss it in the SWG meeting tomorrow.

The definition for the ISO version would then be discussed with the ISO experts. If they (or the ISO Central Secretariat) would prefer another definition, we can simply use different definitions in the OGC and ISO versions of the standard.

**Comment by cportele** [comment]
Decision in the meeting 2020-03-03: Change the definition in 1.0.1 as outlined above.

The harmonization across the organizations (OGC, ISO/TC 211, W3C) will need more time and work and we will contribute to the discussion.

**Comment by heidivanparys** [comment]
As promised during the last meeting, @aidig and I edited a document elaborating on the concept "dataset". It is currently in the form of a draft OGC Discussion Paper, see https://github.com/heidivanparys/discussion_paper_dataset/releases/tag/v20200306

@cportele @ReesePlews

**Comment by cportele** [comment]
Thanks @heidivanparys 👍

**Comment by heidivanparys** [comment]
FYI: Discussion regarding "dataset", and whether "dataset" in DCAT has such broad meaning that software also is a dataset in the DCAT-dataset-sense, is going on in w3c/dxwg#1221 .

Note that the scope of the draft discussion paper mentioned above was deliberately set to having "dataset" in scope and "data" out of scope, however, w3c/dxwg#1221 is still relevant in the broader discussion.
